src/yaml.cpp

```diff
@@ -106,31 +107,49 @@ estd::result<YAML::Node, estd::error> readYamlFile(std::string const & path) {
 }
 
 YamlResult<void> mergeYamlNodes(YAML::Node & map_a, YAML::Node map_b) {
-	// Check if the arguments are maps.
-	if (!map_a.IsMap() && !map_a.IsNull()) {
-		return YamlError{"tried to merge into a YAML node that is not a map"};
-	}
-	if (!map_b.IsMap() && !map_b.IsNull()) {
-		return YamlError{"tried to merge from a YAML node that is not a map"};
-	}
-
 	for (YAML::const_iterator iterator = map_b.begin(); iterator != map_b.end(); iterator++) {
+		
 		std::string key = iterator->first.as<std::string>();
 		YAML::Node value = iterator->second;
-		if (value.IsMap() && map_a[key]) {
+	    if (value.IsMap() && map_a[key]) {
```

Comment:
This is indented with spaces :scream:

I noticed a few more whitespace errors, so could you configure your editor to display whitespace? And be sure to listen to git if it tells you your commit has whitespace errors (and fix them :p ).

Reply:
Updated.